In [1]:
from pint import UnitRegistry
import math
u = UnitRegistry()

# Problema de Detectores (de mi clase)

Se cuenta con un dosímetro [Arrow Tech Inc Model 138](https://www.dosimeter.com/direct-reading-dosimeters/direct-reading-dosimeter-w138-0-200mr-with-sapphire-window/) sometido a una fuente emisora de fotones de 200 keV, para los cuales presenta una eficiencia de 1.8%. Suponer que su volumen activo es una quinta parte de su volumen total, y que su capacidad es de 75 pF.

![Dosimetro](imagenes/detectores/dosimetro-arrow-138.jpg)

1. Calcular la caída de voltaje que se produce al recibir una dosis de 1.3 mGy.
2. Se carga inicialmente el dosímetro con 10 V. ¿Cuántos fotones habrán atravesado el dosímetro para que se descargue a la mitad?
3. ¿A qué dosis corresponderá esa descarga?
4. ¿Es necesario corregir la lectura por las condiciones ambientales?


# Parte 1

## Calcular la caída de voltaje que se produce al recibir una dosis de 1.3 mGy.

El dosímetro es un capacitor, cuya variación de voltaje cumple con la siguiente ecuación:
    
$$\Delta q = C \cdot \Delta V$$

La capacitancia del dosímetro es dato del problema. Para responder hay que encontrar la cantidad de carga producida por esa dosis. Para ello partimos de la definición de exposición:

$$X = \frac{q}{m}$$

Además sabemos que hay una relación fija entre exposición y dosis para cada material. En aire:

$$D_{aire} = \left( \frac{\bar W}{e} \right) _{aire} \cdot X = f \cdot X$$

donde $$f = \left( \frac{\bar W}{e} \right) _{aire} = 33.97 \left[ \frac{J}{C} \right]$$

Podemos expresar:

$$q = X \cdot m =  \frac{D_{aire}}{f} \cdot m$$

Finalmente tenemos la ecuación para la caída de voltaje:
    
$$\Delta V = \frac{q}{C} = \frac{D_{aire} \cdot m}{f \cdot C} $$

La masa de aire que se ioniza en la zona activa del dosimetro se puede calcular considerando el volumen activo del dosímetro, que es un quinto del volumen del cilindro:

$$\text{Vol} = \frac{1}{5} \cdot \pi r^2 h = \frac{1}{5} \cdot \frac{\pi d^2 h}{4}$$

$$ m = \rho_{aire} \cdot \text{Vol}$$

Empezamos los cálculos resolviendo el valor de la masa de aire:

In [2]:
d = 1.5 * u('cm')    # diámetro (de la hoja de datos del detector)
h = 12.4 * u('cm')   # longitud (de la hoja de datos del detector)

dens_aire = 1.205E-03 * u('g/cm^3')    # dato (por ejemplo figura en NIST)

Vol = (1/5) * math.pi * d**2 * h / 4 

m = dens_aire * Vol
m.to('mg')

5.280938710868102 <Unit('milligram')>

Luego calculamos la caida de voltaje que se produce:

In [3]:
Daire = 1.2 * u('mGy')
f = 33.97 * u('J/C')
C = 75 * u('pF')
deltaV = Daire * m / (f * C)
deltaV

2.4873423424754086e-06 <Unit('coulomb * gram * milligray / joule / picofarad')>

In [4]:
deltaV.to('V')

2.4873423424754084 <Unit('volt')>

# Parte 2

## Se carga inicialmente el dosímetro con 10 V. ¿Cuántos fotones habrán atravesado el dosímetro para que se descargue a la mitad?

De la ecuación del capacitor podemos despejar la cantidad de carga depositada para generar una variación de 5V:

$$\Delta q = C \Delta V$$

Para calcular la cantidad de cargas producidas, dividimos la carga total depositada por la carga elemental:

$$n = \frac{\Delta q}{q_e} = \frac{C \Delta V}{q_e}$$

donde $q_e = 1.602176634×10^{−19} C$.

In [5]:
deltaV = 5 * u('V')
qe = 1.602176634e-19 * u('coulomb')
n = C * deltaV / qe
n.to('')

2340565902.9227858 <Unit('dimensionless')>

Para saber cuántos fotones deben interactuar para liberar esa cantidad de cargas, es preciso determinar cuánta carga libera cada fotón. Sabiendo que para producir una ionización en aire se necesitan 33,97 eV podemos calcular cuántos iones produce un fotón de 200 keV:

$ionesPorFoton = \frac{200 keV}{33.97 eV}$

In [6]:
ionesPorFoton = 200 * u('keV') / (33.97 * u('eV'))
ionesPorFoton.to('')

5887.547836326171 <Unit('dimensionless')>

Ahora sí podemos calcular la cantidad de fotones de 200 keV necesarios para generar los "n" iones que producen una caída de 5V en el capacitor:

In [7]:
nroFotones = n / ionesPorFoton
nroFotones.to('')

397545.1186114352 <Unit('dimensionless')>

Como la eficiencia del detector para los fotones de 200 keV es del 1.8%, estos fotones son los que efectivamente interactuaron. Para calcular el total de fotones que atraviesan el detector consideramos la siguiente relación:

|porcentaje| cantidad|
|:---:|:---:|
| 1.8% | 397545 fotones |
| 100% | x fotones |

In [8]:
nroTotalFotones = nroFotones * 100 / 1.8
nroTotalFotones.to('')

22085839.92285751 <Unit('dimensionless')>

# Parte 3


## ¿A qué dosis corresponderá esa descarga?

Recordamos que la dosis es la energía por unidad de masa. El total de energía depositada es la correspondiente al total de energía que entregan los fotones que interactúan con el detector:

In [9]:
Efoton = 200 * u('keV')
energiaEntregada = nroFotones * Efoton
energiaEntregada.to('J')

1.2738749999999997e-08 <Unit('joule')>

La masa ya la tenemos calculada en la primera parte. Por defenición, la dosis se calcula como:

In [10]:
D = energiaEntregada / m
D.to('mGy')

2.4122131873607655 <Unit('milligray')>